src/server/verification/verificationAPI.js

```diff
@@ -282,22 +282,35 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
 
       //merge user details for use by mautic
       let userRec: UserRecord = _.defaults(body.user, user)
-      const savedEmail = userRec.email
+      const currentEmail = user.email
+      const tempMauticId = user.otp && user.otp.tempMauticId
 
       if (conf.allowDuplicateUserData === false && (await storage.isDupUserData({ email }))) {
         return res.json({ ok: 0, error: 'Email already exists, please use a different one' })
       }
-      if (!user.mauticId || user.email !== body.email) {
-        //first time create contact for user in mautic
+
+      if ((!user.mauticId && !tempMauticId) || (currentEmail && currentEmail !== email)) {
         const mauticContact = await Mautic.createContact(userRec)
-        userRec.mauticId = mauticContact.contact.fields.all.id
+
+        userRec.otp = {
```

Comment:
dont over use ...
its simply userRec.otp.tempMauticId=

Reply:
@sirpy I did like you suggesting at the first time. But during the test, I found that there could be a case when otp is undefined, so the error will occur. That is why I rewrite it so.